antarest/study/output/file_output_storage.py

```
@@ -0,0 +1,376 @@
+
+        stopwatch = StopWatch()
+        if not path_output_zip.exists():
+            archive_dir(path_output, target, archive_format=ArchiveFormat.ZIP)
+        stopwatch.log_elapsed(lambda x: logger.info(f"Output {output_id} from study {study_id} exported in {x}s"))
+
+    @override
+    def create_output_download(
+        self, study_id: str, output_id: str, data: StudyDownloadDTO, filetype: ExportFormat, target: Path
+    ) -> None:
+        """
+        For the "download" feature of the API.
+        """
+        study_outputs = self._outputs_provider.get_outputs(study_id)
+        stopwatch = StopWatch()
+        matrix = StudyDownloader.build(
+            study_outputs.get_file_study(),
+            output_id,
+            data,
+        )
+        stopwatch.log_elapsed(lambda x: logger.info(f"Study {study_id} filtered output {output_id} built in {x}s"))
+        StudyDownloader.export(matrix, filetype, target)
+        stopwatch.log_elapsed(lambda x: logger.info(f"Study {study_id} filtered output {output_id} exported in {x}s"))
+
+    @override
+    def output_exists(self, study_id: str, output_id: str) -> bool:
+        """Check if a study output exists."""
+        study_outputs = self._outputs_provider.get_outputs(study_id)
+        if self.is_output_archived(study_id, output_id):
+            output_path = study_outputs.outputs_path / f"{output_id}{ArchiveFormat.ZIP}"
+            return output_path.exists()
+        else:
+            output_path = study_outputs.outputs_path / output_id
+            return output_path.is_dir()
+
+    @override
+    def is_output_archived(self, study_id: str, output_id: str) -> bool:
+        """Check if a study output is archived."""
+        study_outputs = self._outputs_provider.get_outputs(study_id)
+        output_path = study_outputs.outputs_path / output_id
+        return is_output_archived(output_path)
+
+    @override
+    def archive_study_output(self, study_id: str, output_id: str) -> bool:
```

Here we return a bool but we don't use it i think. And same remark we should remove the semi-zip in case of a failure instead of returning False